designs/developer/4.8/upload-jar-file/index.md

```diff
@@ -0,0 +1,59 @@
+---
+parent: Developer
+version: 4.8
+---
+
+# Uploading a JAR file
+
+## Overview
+
+In 4.8, the ability to upload a JAR file to create a Spring Boot application was added to the Developer Console. There are couple different ways a user can upload a JAR file to OpenShift. They can drag and drop their JAR file directly onto the Topology, go to the +Add page to select the 'Upload JAR file' option, or right click in the Topology to access the Add to Project menu and select "Upload JAR file".
+
+The user can also go back and make edits to their application by re-uploading or updating the attached JAR file.   
+
```

Comment:
I think this sentence needs a rewording. Once they have uploaded a Jar file, if the user wants to edit that .Jar file, they would edit it locally and then re-upload it, right? Is there any mechanism to scan the file to see if it should replace something?

"After the .Jar file has been uploaded,  the user can re-upload the file when they have edits. Once the file is uploaded again, it will replace the original file."

@lwrigh do we ask them if they want to replace it? Do we check the name or parse the file before we do a replacement?

Reply:
Yes that is correct from my understanding.  If the user wants to make edits to the existing .Jar file, they would need to edit it locally and then re-upload it. I don't think there is any mechanism that scans the file and brings in the new edits. I think it's an entirely new upload each time the user re-uploads the .Jar file with new edits. 

There is currently no confirmation before the user hits the 'Save' button on the edit Jar file form. The user has to go through the file upload process each time they make an edit to the .Jar file and re-upload it. I don't think the back end currently does anything to check the name or parse the file. The user can re-upload a .Jar file with the same name as the old .Jar file and re-upload it without any naming errors. 

I'll make updates to this paragraph so that interaction is more clear.